src/site/content/en/progressive-web-apps/customize-install/index.md

```diff
@@ -30,68 +20,101 @@ it's possible to provide your own custom install flow, directly within your app.
   remains committed to supporting them, and has no plans to remove or deprecate
-  support. Google's Web DevRel team continues to recommend using them to provide a customized
-  install experience.
+  support. Google's Web DevRel team continues to recommend using them to
+  provide a customized install experience.
 {% endAside %}
 
-### Listen for the `beforeinstallprompt` event {: #beforeinstallprompt }
+### Page handles the `beforeinstallprompt` event {: #beforeinstallprompt }
+
+If your Progressive Web App meets the required [installation criteria][criteria],
+the browser fires a `beforeinstallprompt` event. The `beforeinstallprompt`
+event tells the PWA that it's installable, and that it can show the in-app
+install experience.
+
+{% Aside 'caution' %}
+To prevent confusion with your users, don't show your own custom install
+experience until the `beforeinstallprompt` event has been fired.
+{% endAside %}
 
-If your Progressive Web App meets the required [installation criteria](/install-criteria/),
-the browser fires a `beforeinstallprompt` event. Save a reference to the
-event, and update your user interface to indicate that the user can install
-your PWA. This is highlighted below.
+Save a reference to the event, and update your user interface to indicate
+that the user can install your PWA. And, so we can measure the effectiveness
+of the install flow, log it to analytics.
 
 ```js
 // Initialize deferredPrompt for use later to show browser install prompt.
 let deferredPrompt;
+// Initialize installSource, used by analytics to identify which in-app
+// install element is clicked.
+let installSource;
 
 window.addEventListener('beforeinstallprompt', (e) => {
-  // Prevent the mini-infobar from appearing on mobile
+  // Optionally, prevent the mini-infobar from appearing on mobile.
   e.preventDefault();
   // Stash the event so it can be triggered later.
   deferredPrompt = e;
-  // Update UI notify the user they can install the PWA
+  // Update UI notify the user they can install the PWA.
   showInstallPromotion();
   // Optionally, send analytics event that PWA install promo was shown.
-  console.log(`'beforeinstallprompt' event was fired.`);
+  ga('send', 'event', {
+    eventCategory: 'pwa-install',
+    eventAction: 'promo-shown',
+    nonInteraction: true,
+  });
 });
 ```
 
+For consistency, all the analytics install events use the same category,
+`pwa-install`. In this case, the action is `promo-shown`. And set
+`non-interaction` to `true`, since this event wasn't generated by a user
+action.
+
 {% Aside %}
 There are many different [patterns](/promote-install/) that you can use to
 notify the user your app can be installed and provide an in-app install
 flow, for example, a button in the header, an item in the navigation menu,
 or an item in your content feed.
 {% endAside %}
 
-### In-app installation flow {: #in-app-flow }
+### Page handles click on in-app install element {: #click-in-app-install }
```

Comment:
I'm having trouble processing this title. Part of the problem is two sequential prepositions (on-in). Grammatically it's fine. A prepositional phrase can function as an adjective, which is what 'in-app' is doing.